### DISCUSSION TITLE
ScriptReference/EditorGUI.EndChangeCheck

### DISCUSSION THREAD

**Discussion by docscomments** [Developer-Notes-Extension/Unity-Notes#199] [Notes]
# ScriptReference/EditorGUI.EndChangeCheck

https://docs.unity3d.com/ScriptReference/EditorGUI.EndChangeCheck.html

**Comment by mvi** [comment]
Note that Begin/EndChangeCheck can be nested. For example:  
  

```csharp
string foo = "";  
int bar = 0;  
  
EditorGUI.BeginChangeCheck();  
  
foo = EditorGUILayout.TextField(foo);  
  
EditorGUI.BeginChangeCheck();  
  
bar = EditorGUILayout.IntField(bar);  
  
if (EditorGUILayout.EndChangeCheck())  
{  
    Debug.Log("Bar changed");  
}  
  
if (EditorGUILayout.EndChangeCheck())  
{  
    Debug.Log("Foo or Bar changed");  
}
```
  
Also, as mentioned in the docs this is based on `GUI.changed` going from `false` at the time of `BeginChangeCheck` to `true` at the time of `EndChangeCheck`. Built in Unity controls will mark this flag already, but if you need it with your own controls or with a button you can set `GUI.changed = true;` yourself.
